pkg/inventory/inventory.go

```diff
@@ -0,0 +1,215 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inventory
+
+import (
+	"encoding/json"
+
+	"sigs.k8s.io/kustomize/pkg/resid"
+)
+
+// A Refs is a map from an Item string to a list of Items
+// that it is referred
+type Refs map[resid.ItemId][]resid.ItemId
```

Comment:
```
Refs is a reference map.  Each key is the id
of a k8s resource, and each value is a list of
object ids that refer back to the object in the
key.

For example, the key could correspond to a
ConfigMap, and the list of values might include
several different Deployments that get data from
that ConfigMap (and thus refer to it).

References are important in inventory management
because one may not delete an object before all
objects referencing it have been removed.
```